mitiq/zne/inference.py

```diff
@@ -1511,6 +1513,141 @@ def __eq__(self, other: Any) -> bool:
+        shot_list: Optional sequence of integers corresponding to the number
+                   of samples taken for each expectation value. If this
+                   argument is explicitly passed to the factory, it must have
+                   the same length of scale_factors and the executor function
+                   must accept "shots" as a valid keyword argument.
+
+    Raises:
+        ValueError: If data is not consistent with the extrapolation model.
+    Note:
+        RichardsonFactory and LinearFactory are special cases of PolyFactory.
+    """
+
+    @staticmethod
+    def _exp_ansatz(
+        a: float,
+        b: float,
+        c: float,
+        scale_factor: float
+    ) -> float:
+        """
+        Calculates the expecation given a scale factor and model
+        parameters.
+
+        Args:
+            a: Model parameter.
+            b: Model parameter.
+            c: Model parameter.
+            scale_factors: The array of noise scale factors.
+
+        Returns:
+            The expected value according to the model parameters and
+            scale factor.
+        """
+
+        return a + b * np.exp(-1.0 * c * scale_factor)
+
+    @staticmethod
+    def extrapolate(
+        scale_factors: Sequence[float],
+        exp_values: Sequence[float],
+        full_output: bool = False,
+    ) -> Union[
+
+        float,
+    ]:
+        """Static method which evaluates an exponential extrapolation to the
+        zero-noise limit using Bayesian inference.
+
+        Args:
+            scale_factors: The array of noise scale factors.
+            exp_values: The array of expectation values.
+            full_output: If False (default), only the zero-noise limit is
+                returned. If True, additional information about the
+                extrapolated limit is returned too.
+        Returns:
+            zne_limit: The extrapolated zero-noise limit. If "full_output"
+                is False (default value), only this parameter is returned.
+            opt_params: The parameter array of the best fitting model.
+            zne_curve: The callable function which best fit the input data.
+                It maps a real noise scale factor to a real expectation value.
+                It is equal "zne_limit" when evaluated at zero.
+
+        Note:
+            This method computes the zero-noise limit only from the information
+            contained in the input arguments. To extrapolate from the internal
+            data of an instantiated Factory object, the bound method
+            ".reduce()" should be called instead.
+        """
+        with pm.Model():
+            """
+            We assumme that the priors for the model parameters is a uniform
+            distribution with an upper limit 1 and a lower limit 0,
+            except for b with a lower limit -1.
+            """
+            a = pm.Uniform('a', 0, 1)
+            b = pm.Uniform('b', -1, 1)
+            c = pm.Uniform('c', 0, 1)
```

Comment:
Are all values inside this cube feasible? You may need to postselect to prevent values that lead to probabilities outside [0, 1].